### PR TITLE
dev-libs/re2: version bump to 0_p20160401

### PR DESCRIPTION
**Pull request by lluixhi**
Rewrote it to use EAPI 6 as well.

Gentoo-Bug: https://bugs.gentoo.org/579136
Gentoo-Bug: https://bugs.gentoo.org/571156
Gentoo-Bug: https://bugs.gentoo.org/572684

### CODE REVIEW

**Comment by lluixhi** [comment]
Not sure who's maintaining this. @floppym @asarubbo?

**Comment by lluixhi** [comment]
By the way, I got the makefile patch upstreamed.
https://github.com/google/re2/commit/908a6fa4f680002884065252fb8dee96a260cc4d

**Comment by floppym** [comment]
Sorry, will try to review soon.

File: dev-libs/re2/re2-0_p20160401.ebuild [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+PATCHES=( "${FILESDIR}/${PN}-${RE2_VER}-makefile.patch" )
+DOCS=( "AUTHORS" "CONTRIBUTORS" "README" "doc/syntax.txt" )
+HTML_DOCS=( "doc/syntax.html" )
+
+src_prepare() {
+	default
+
+	if use icu; then
+		sed -i '7,8 s/# //' Makefile
```
Comment (rindeal) [inline dev-libs/re2/re2-0_p20160401.ebuild]: `use icu && ... || die`

File: dev-libs/re2/re2-0_p20160401.ebuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	fi
+}
+
+src_install() {
+	emake DESTDIR="${ED}" prefix=usr libdir=usr/$(get_libdir) install
```
Comment (rindeal) [inline dev-libs/re2/re2-0_p20160401.ebuild]: `s|usr|${EROOT}usr|g` + some spare `"` would be ok

File: dev-libs/re2/re2-0_p20160401.ebuild (hunk 1 above)
Comment (floppym) [inline dev-libs/re2/re2-0_p20160401.ebuild]: I don't think the conditional block is necessary at all really.

File: dev-libs/re2/re2-0_p20160401.ebuild (hunk 2 above)
Comment (floppym) [inline dev-libs/re2/re2-0_p20160401.ebuild]: EROOT is definitely wrong.

Comment (rindeal) [inline dev-libs/re2/re2-0_p20160401.ebuild]: And what is EROOT actually for?

Comment (floppym) [inline dev-libs/re2/re2-0_p20160401.ebuild]: EROOT is mainly used to refer to installed files in pkg_postinst.

**Pull request closed without merging**